src/memory.rs

```diff
@@ -446,9 +452,24 @@ impl<'a, 'tcx> Memory<'a, 'tcx> {
 /// Reading and writing
 impl<'a, 'tcx> Memory<'a, 'tcx> {
     pub fn freeze(&mut self, alloc_id: AllocId) -> EvalResult<'tcx, ()> {
+        // FIXME: the comment is wrong and do we really still need this check?
```

Comment:
I think we don't need it anymore... the problem, if I recall correctly, involved the ZST alloc getting frozen for one constant and then later "mutated" for an unrelated value, but I think we no longer treat zero-sized writes as mutations.